### PR TITLE
[veSync] Improve recognition and device support

### PR DESCRIPTION
**Pull request by dag81**
This provides improved support for device recognition, ensuring devices from different regions are identified including a new mist model. Logic adjustments have been made to group devices into device family's to keep the code base maintainable, in line with the groups defined within PyVeSync.

Testing has been requested in this thread: https://community.openhab.org/t/vesync-binding-help-testing-requested/144175

Signed-off-by: David Goodyear <david.goodyear@gmail.com>

### CODE REVIEW

**Comment by dag81** [comment]
Converted to draft to add annotations missed, but seen when porting back to 3.4.X

**Comment by dag81** [comment]
Hi all, any chance of getting a review / feedback on this? As would like to get this in v4 prior to back-porting.

File: bundles/org.openhab.binding.vesync/README.md [hunk 1]
```diff
@@ -208,6 +208,21 @@ Number:Dimensionless LoungeAHTarget            "Lounge Air Humidifier Target Hum
 Number:Dimensionless LoungeAHMistLevel         "Lounge Air Humidifier Mist Level"                             { channel="vesync:airHumidifier:vesyncServers:loungeHumidifier:mistLevel" }
 ```
 
+#### Air Humidifier Oasis Mist Smart Model
+
+```
```
Comment (jlaur) [inline bundles/org.openhab.binding.vesync/README.md:213]:
```suggestion
```java
```

File: bundles/org.openhab.binding.vesync/README.md [hunk 2]
```diff
@@ -295,6 +310,23 @@ Frame {
 }
 ```
 
+#### Air Humidifier Oasis Mist Smart Model
+
+```
```
Comment (jlaur) [inline bundles/org.openhab.binding.vesync/README.md:315]:
```suggestion
```perl
```

**Review by jlaur — APPROVED** [review]
Many thanks, LGTM. Only two minor comments for correct syntax highlighting (by convention).

File: bundles/org.openhab.binding.vesync/README.md (hunk 2 above)
Comment (dag81) [reply bundles/org.openhab.binding.vesync/README.md:315]: Thank you @jlaur, I hadn't seen that in any of the other bindings I've looked at. After searching I see its common :) its now updated to the convention for the sitemap examples to include the perl tag.

File: bundles/org.openhab.binding.vesync/README.md (hunk 1 above)
Comment (dag81) [reply bundles/org.openhab.binding.vesync/README.md]: @jlaur - Thank you. I'm surprised by this one as, my IDE def, does not like non java code having that marker. Like before on searching I found it in many other binding's- so have aligned it to the convention. Thanks again.

**Comment by dag81** [comment]
@jlaur Thank you for pointing out more bits I hadn't spotted before :) 🤞 its all good now the final changes should be in the branch now, going through the checks.

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**